test/ibmq/test_jobmanager.py

```diff
@@ -0,0 +1,216 @@
+        print(report)
+        for job in jobs:
+            self.assertIn(job.job_id(), report)
+
+    @requires_provider
+    def test_skipped_result(self, provider):
+        """Test one of jobs has no result."""
+        backend = provider.get_backend('ibmq_qasm_simulator')
+        max_circs = backend.configuration().max_experiments
+
+        circs = []
+        for _ in range(max_circs+2):
+            circs.append(self._qc)
+        self._jm.run(circs, backend=backend)
+        jobs = self._jm.jobs()
+        cjob = jobs[0]
+        cancelled = False
+        for _ in range(2):
+            # Try twice in case job is not in a cancellable state
+            try:
+                cancelled = cjob.cancel()
+                if cancelled:
+                    break
+            except JobError:
+                pass
+
+        results = self._jm.result()
+        statuses = self._jm.status()
+        if cancelled:
+            self.assertTrue(statuses[0] is JobStatus.CANCELLED)
+            self.assertIsNone(
+                results[0], "Job {} cancelled but result is not None.".format(cjob.job_id()))
+        else:
+            self.log.warning("Unable to cancel job %s", cjob.job_id())
+
+    @requires_provider
+    def test_skipped_status(self, provider):
+        """Test one of jobs has no status."""
+        backend = provider.get_backend('ibmq_qasm_simulator')
+
+        circs = []
+        for _ in range(2):
+            circs.append(self._qc)
+        self._jm.run(circs, backend=backend, max_experiments_per_job=1)
+        jobs = self._jm.jobs()
+        jobs[1]._job_id = 'BAD_ID'
+        statuses = self._jm.status()
+        self.assertIsNone(statuses[1])
+
+    @requires_provider
+    def test_job_qobjs(self, provider):
+        """Test retrieving qobjs for the jobs."""
+        backend = provider.get_backend('ibmq_qasm_simulator')
+
+        circs = []
+        for _ in range(2):
+            circs.append(self._qc)
+        self._jm.run(circs, backend=backend, max_experiments_per_job=1)
+        jobs = self._jm.jobs()
+        self._jm.result()
+        for i, qobj in enumerate(self._jm.qobj()):
+            rjob = provider.backends.retrieve_job(jobs[i].job_id())
+            self.assertDictEqual(qobj.__dict__, rjob.qobj().__dict__)
+
+    @run_on_staging
+    def test_error_message(self, provider):
+        """Test error message report."""
+        backend = least_busy(provider.backends(simulator=False))
+
+        bad_qc = copy.deepcopy(self._qc)
+        circs = [transpile(self._qc, backend=backend), bad_qc]
+        self._jm.run(circs, backend=backend, max_experiments_per_job=1, skip_transpile=True)
+
+        results = self._jm.result()
+        self.assertIsNone(results[1])
+
+        error_report = self._jm.error_message()
+        self.assertIsNotNone(error_report)
+        print("JobManager.error_message(): \n{}".format(error_report))
+        print("\nJobManager.report(): \n{}".format(self._jm.report()))
```

Comment:
Can you do a pass for the `print`s, which seem to be leftovers?

Reply:
They were intentional to allow someone to "eyeball" them in the Travis log. But perhaps they just pollute the log. I'll remove them.